algorithms/backtrack/generate_parenthesis.py

```diff
@@ -13,20 +13,21 @@
 ]
 """
 
-def gen_parenthesis(n:"int")->"List[str]":
+
+def generate_parenthesis(n: int)->list[str]:
+    def add_pair(res, s, left, right):
+        if left == 0 and right == 0:
+            res.append(s)
+            return
+        if right > 0:
+            add_pair(res, s+")", left, right-1)
+        if left > 0:
+            add_pair(res, s+"(", left-1, right+1)
+
     res = []
     add_pair(res, "", n, 0)
     return res
 
-def add_pair(res, s, left, right):
-    if left == 0 and right == 0:
-        res.append(s)
-        return
-    if right > 0:
-        add_pair(res, s+")", left, right-1)
-    if left > 0:
-        add_pair(res, s+"(", left-1, right+1)
-
 
-if __name__=="__main__":
-    print(gen_parenthesis(3))
+if __name__ == "__main__":
+    print(generate_parenthesis(3))
```

Comment:
I think we should remove this `if __name__ == "__main__"` for consistency with the reet of the files.

Reply:
gotcha